packages/swapper/src/swappers/zrx/approvalNeeded/approvalNeeded.ts

```diff
@@ -0,0 +1,77 @@
+import { BigNumber } from 'bignumber.js'
+import { AxiosResponse } from 'axios'
+import { ChainAdapter } from '@shapeshiftoss/chain-adapters'
+import {
+  ApprovalNeededInput,
+  ApprovalNeededOutput,
+  ChainTypes,
+  QuoteResponse
+} from '@shapeshiftoss/types'
+import {
+  AFFILIATE_ADDRESS,
+  APPROVAL_BUY_AMOUNT,
+  APPROVAL_GAS_LIMIT,
+  DEFAULT_ETH_PATH,
+  DEFAULT_SLIPPAGE
+} from '../utils/constants'
+
+import { ZrxSwapperDeps } from '../ZrxSwapper'
+import { zrxService } from '../utils/zrxService'
+import { getERC20Allowance } from '../utils/helpers/helpers'
+import { erc20AllowanceAbi } from '../utils/abi/erc20Allowance-abi'
+
+export async function approvalNeeded(
+  { adapterManager, web3 }: ZrxSwapperDeps,
+  { quote, wallet }: ApprovalNeededInput
+): Promise<ApprovalNeededOutput> {
+  const { sellAsset } = quote
+
+  if (sellAsset.symbol === 'ETH' || sellAsset.chain !== ChainTypes.Ethereum) {
```

Comment:
i would prob break this out, return false if symbol is eth, and throw if chain is not eth